docs/program/_index.md

```diff
@@ -3,8 +3,278 @@ title: "Program a Robot"
+      })),
+  )
+  if err != nil {
+      logger.Fatal(err)
+  }
+  defer robot.Close(context.Background())
+  logger.Info("Resources:")
+  logger.Info(robot.ResourceNames())
+}
+```
+
+{{% /tab %}}
+{{% tab name="TypeScript" %}}
+
+{{< alert title="Note" color="note" >}}
+The TypeScript SDK currently only supports building web browser apps.
+{{< /alert >}}
+
+```ts {class="line-numbers linkable-line-numbers"}
+import { Client, createRobotClient, RobotClient } from '@viamrobotics/sdk';
+
+async function connect() {
+  // You can remove this block entirely if your robot is not authenticated.
+  // Otherwise, replace with an actual secret.
+  const secret = '<SECRET>';
+  const credential = {
+    payload: secret,
+    type: 'robot-location-secret',
+  };
+
+  // Replace with the host of your actual robot running Viam.
+  const host = "<HOST>";
+
+  // Replace with the signaling address. If you are running your robot on Viam,
+  // it is most likely https://app.viam.com:443.
+  const signalingAddress = 'https://app.viam.com:443';
+
+  const iceServers = [{ urls: 'stun:global.stun.twilio.com:3478' }];
+
+  return createRobotClient({
+    host,
+    credential,
+    authEntity: host,
+    signalingAddress,
+    iceServers
+  });
+}
+
+async function main() {
+  // Connect to client
+  let client: Client;
+  try {
+    client = await connect();
+    console.log('connected!');
+
+    let resources = await client.resourceNames();
+    console.log('Resources:');
+    console.log(resources);
+  } catch (error) {
+    console.log(error);
+    return;
+  }
+}
+
+main();
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+{{< alert title="Note" color="note" >}}
+The C++ SDK is currently in alpha.
+{{< /alert >}}
+
+```cpp {class="line-numbers linkable-line-numbers"}
+# include <string>
+# include <vector>
+
+# include <boost/optional.hpp>
+
+# include <viam/api/common/v1/common.pb.h>
+# include <viam/api/robot/v1/robot.grpc.pb.h>
+
+# include <viam/sdk/robot/client.hpp>
+# include <viam/sdk/components/camera/client.hpp>
+
+using namespace viam::sdk;
+
+int main() {
+  std::string host("ADDRESS FROM THE VIAM APP");
+  DialOptions dial_opts;
+  Credentials credentials("SECRET FROM THE VIAM APP");
+  dial_opts.set_credentials(credentials);
+  boost::optional<DialOptions> opts(dial_opts);
+  Options options(0, opts);
+
+  auto robot = RobotClient::at_address(host, options);
+
+  std::cout << "Resources:\n";
+  for (const ResourceName& resource: *robot->resource_names()) {
+    std::cout << resource.namespace_() << ":" << resource.type() << ":"
+              << resource.subtype() << ":" << resource.name() << "\n";
+  }
+
+  return 0;
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+{{% /expand%}}
+
+Save this file on your development machine with the file type of your preferred SDK.
+
+Edit the boilerplate code by [adding control logic](/program/sdks/) to make a client application that connects to your robot and controls it in the way you want.
```

Comment:
I think this sentence actually should come later.  Typically one would run the sample code-as is first to ensure requirements are met and connectivity works, then they'd alter it to suit their needs

Reply:
@mcvella That is a good point re. why that would be needed. Done!